webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactStorage.scala

```diff
@@ -196,17 +205,24 @@ object StorageChangeEventSave {
       def update(nfa: NodeFact, nfb: NodeFact, attrs: SelectFacts) = SelectFacts.merge(nfa, Some(nfb))(attrs.invert)
 
       (e, b) match {
-        case (Noop(_, _), x)                                  => x
-        case (x, Noop(_, _))                                  => x
-        case (Created(a1, _), Created(a2, attrs))             => Created(update(a1, a2, attrs), attrs)
-        case (Created(a1, _), Updated(ob, nb, attrs))         => Updated(ob, update(a1, nb, attrs), attrs)
-        case (Updated(oa, na, _), Created(b1, attrs))         => Updated(oa, update(na, b1, attrs), attrs)
-        case (Updated(ob1, nb1, _), Updated(ob2, nb2, attrs)) =>
+        case (Noop(id, ata), Noop(_, atb))                => Noop(id, ata.max(atb))
+        case (Noop(_, ata), Created(ob, atb))             => Created(ob, ata.max(atb))
```

Comment:
correcting the `SelectFacts` in change event composition: we want to report if a change happened in first or in second event, not ignore half of them.